geo/src/algorithm/k_nearest_concave_hull.rs

```diff
@@ -0,0 +1,393 @@
+    multipoint.convex_hull()
+}
+
+fn get_next_k(curr_k: u32) -> u32 {
+    max(curr_k + 1, ((curr_k as f32) * K_MULTIPLIER) as u32)
+}
+
+fn adjust_k(k: u32) -> u32 {
+    max(k, 3)
+}
+
+fn get_first_point<T>(point_set: &rstar::RTree<Point<T>>) -> &Point<T>
+where T: GeoFloat + RTreeNum
+{
+    let mut min_y = Float::max_value();
+    let mut result = point_set.iter().next().expect("We checked that there are more then 3 points in the set before.");
+
+    for p in point_set.iter() {
+        if p.y() < min_y {
+            min_y = p.y();
+            result = p;
+        }
+    }
+
+    result
+}
+
+fn get_nearest_points<'a, T>(dataset: &'a rstar::RTree<Point<T>>, base_point: &Point<T>, candidate_no: u32) -> Vec<&'a Point<T>>
+where T: GeoFloat + RTreeNum
+{
+    dataset.nearest_neighbor_iter(base_point).take(candidate_no as usize).collect()
+}
+
+fn sort_by_angle<'a, T>(points: &'a mut Vec<&Point<T>>, curr_point: &Point<T>, prev_angle: T)
+where T: GeoFloat
+{
+    points.sort_by(|a, b| {
+        let mut angle_a = get_angle(&[curr_point, a]) - prev_angle;
+        let mut angle_b = get_angle(&[curr_point, b]) - prev_angle;
+        if angle_a < T::zero() { angle_a = angle_a + two_pi(); }
+        if angle_b < T::zero() { angle_b = angle_b + two_pi(); }
+        angle_a.partial_cmp(&angle_b).unwrap().reverse()
+    });
+}
+
+fn intersects<T>(hull: &Vec<Point<T>>, line: &[&Point<T>; 2]) -> bool
+where T: GeoFloat
+{
+    // This is the case of finishing the contour.
+    if *line[1] == hull[0] { return false; }
+
+    let points = hull.iter().take(hull.len() - 1).map(|x| crate::Coordinate::from(x.clone())).collect();
+    let linestring = LineString(points);
+    let line = crate::Line::new(*line[0], *line[1]);
+    linestring.intersects(&line)
+}
+
+fn get_angle<T>(line: &[&Point<T>; 2]) -> T
+where T: GeoFloat
+{
+    let x1 = line[0].x();
+    let y1 = line[0].y();
+    let x2 = line[1].x();
+    let y2 = line[1].y();
+
+    let a = x2 - x1;
+    let b = y2 - y1;
+    let c = (a*a + b*b).sqrt();
+    if c == T::zero() {
+        return T::zero();
+    }
+
+    let cos = a / c;
+    let mut acos = cos.acos();
+    if y1 > y2 {
+        acos = two_pi::<T>() - acos;
+    }
+
+    acos
+}
+
+fn point_inside<T>(point: &Point<T>, poly: &Polygon<T>) -> bool
+where T: GeoFloat
+{
+    poly.contains(point) || poly.exterior().contains(point)
+}
+
+fn two_pi<T>() -> T
+where T: GeoFloat
+{
+    T::from(std::f64::consts::PI * 2f64).unwrap()
+}
+
+#[cfg(test)]
+mod tests {
+    use test_case::test_case;
+    use crate::point;
+    use super::*;
+    use crate::coords_iter::CoordsIter;
+    use num_traits::FloatConst;
+    use std::f32::consts::PI;
+
+    #[test_case(0., 0., 1., 0., 0.)]
+    #[test_case(0., 0., 0., 1., PI / 2.)]
+    #[test_case(0., 0., -1., 0., PI)]
+    #[test_case(0., 0., 0., -1., PI / 2. * 3.)]
+    #[test_case(0., 0., -1., 1., PI / 4. * 3.)]
+    #[test_case(0., 0., -1., -1., PI / 4. * 5.)]
```

Comment:
This is admittedly kind of nice, but I'd prefer to avoid adding an extra dependency for it.

(Also, curious if, as @lnicola mentioned, we could remove get_angle altogether)